packages/keyring-controller/src/KeyringController.ts

```diff
@@ -658,18 +679,25 @@ export class KeyringController extends BaseController<
    * Adds a new account to the default (first) HD seed phrase keyring.
    *
    * @param accountCount - Number of accounts before adding a new one, used to
+   * @param keyringId - The id of the keyring to add the account to.
    * make the method idempotent.
    * @returns Promise resolving to the added account address.
    */
-  async addNewAccount(accountCount?: number): Promise<string> {
+  async addNewAccount(
+    accountCount?: number,
+    keyringId?: string,
+  ): Promise<string> {
```

Comment:
Similarly to my comment on `verifySeedPhrase`, this seems too broad and would break with any non-HD keyring. I believe that the client could add new accounts using `withKeyring` alone, which makes this change unnecessary